torch/_dynamo/utils.py

```diff
@@ -761,10 +762,24 @@ def clone_inputs(example_inputs):
     return res
 
 
+def skip_frame_if_in_functorch_mode(val: torch.Tensor):
+    try:
+        val.data_ptr()  # will throw for functorch tensors
```

Comment:
functorch is not the only place where one can get a C++ tensor subclass without a data_ptr, but in general we should do what this PR does (graph break on Tensors that don't have data_ptr).

Reply:
Yeah, they should all get graph breaks.